packages/server/src/main/java/software/uncharted/terarium/hmiserver/models/dataservice/project/Project.java

```diff
@@ -79,4 +79,40 @@ public class Project extends TerariumAsset {
 	@Transient
 	@Schema(accessMode = Schema.AccessMode.READ_ONLY)
 	private String userPermission;
+
+	public static Project merge(final Project existingProject, final Project project) {
+
+		// Merge non-transient fields to ensure they are not lost
+		if (project.getId() != null) {
+			existingProject.setId(project.getId());
+		}
+		if (project.getName() != null) {
+			existingProject.setName(project.getName());
+		}
+		if (project.getUserId() != null) {
+			existingProject.setUserId(project.getUserId());
+		}
+		if (project.getDescription() != null) {
+			existingProject.setDescription(project.getDescription());
```

Comment:
These all make sense, but im not sure about description and overviewContent.  If I as a user erase the description on a project (or the overview), will this bring it back?

Reply:
good question. Null should not be updated, but "" should... I will test

Reply:
yeah, empty strings are acceptable.